Common/Data/Encoding/Utf8.h

```diff
@@ -26,6 +26,14 @@ int u8_strlen(const char *s);
 void u8_inc(const char *s, int *i);
 void u8_dec(const char *s, int *i);
 
+// ranges grabbed from https://stackoverflow.com/a/62898106, ignoring the two bogus ranges.
+// there's probably more. Doesn't need to be perfect.
+inline bool CodepointIsProbablyEmoji(uint32_t c) {
+	return (c >= 127744 && c <= 129782) || (c >= 126980 && c <= 127569);
```

Comment:
Personally, I would just assume anything with characters outside the BMP (i.e. c > 0xFFFF) includes emoji.  Unless someone is using Ancient Sumerian or Egyptian Hieroglyphics, that's probably the only reason most people will use non-BMP glyphs.  I guess the other thing is advanced math symbols.

-[Unknown]

Reply:
Yeah, I guess that's the safer way to go...